api/envoy/extensions/transport_socket/capture/v2/capture.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+// Trace capture format for the capture transport socket extension. This dumps plain text read/write
+// sequences on a socket.
+
+package envoy.extensions.transport_socket.capture.v2;
+option go_package = "v2";
+
+import "envoy/api/v2/core/address.proto";
+
+import "google/protobuf/timestamp.proto";
+
+// Connection properties.
+message Connection {
+  // Global unique connection ID for Envoy session. Matches connection IDs used
+  // in Envoy logs.
+  uint64 id = 1;
+  // Local address.
+  envoy.api.v2.core.Address local_address = 2;
+  // Remote address.
+  envoy.api.v2.core.Address remote_address = 3;
+}
+
+// Event in a capture trace.
+message Event {
+  // Timestamp for event.
+  google.protobuf.Timestamp timestamp = 1;
+  message Read {
```

Comment:
What context/direction are these terms referring to? Is the simulating envoy reading from the kernel, or the kernel reading from envoy? Please document clearly.